### PR TITLE
Add search bar + receipt notifications

### PR DESCRIPTION
**Pull request by Sup3Legacy**
Implements a new text input box. One can switch between this box and the chat box using a shortcut (`ALT+Tab` for now). When this box is non-empty, the text is used to search through all channels. 

It is possible to either type in directly a name that then gets matched against the name of every channel or precede it by `@` and have it instead matched against the members names.

I also incorporated the receipt features into this PR as it is becoming kind of a merge mess between these branches.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -113,6 +230,23 @@ pub struct GroupData {
...
+
+    pub fn match_pattern(&self, pattern: &str, hm: &HashMap<Uuid, String>) -> bool {
+        if pattern.is_empty() {
+            return true;
+        }
+        match pattern.chars().next().unwrap() {
+            '@' => self.contains_user(&pattern[1..], hm),
+            _ => self.name.contains(pattern),
+        }
+    }
```
Comment (Sup3Legacy) [inline src/app.rs:248]: This function can be expand to allow for more advanced search patterns.

Comment (Sup3Legacy) [inline src/app.rs:248]: This function can be expand to allow for more advanced search patterns.

**Review by boxdot — APPROVED** [review]
This is huge! Thanks.

**Comment by davepinto** [comment]
Thank you for the feature, but is there a way to configure the key-binding for this (without modifying source code + recompiling)? I'm in the unfortunate situation of needing to use windows for work, which conflicts with the Alt+Tab window switching key-binding, making this feature essentially in-accessible.

**Comment by boxdot** [comment]
How about Ctrl+P?

**Pull request merged**